pandas/tseries/tools.py

```diff
@@ -349,7 +350,11 @@ def _convert_listlike(arg, box, format):
     elif com.is_list_like(arg):
         return _convert_listlike(arg, box, format)
 
-    return _convert_listlike(np.array([ arg ]), box, format)[0]
+    try:
+        return _convert_listlike(np.array([ arg ]), box, format)[0]
+    except ValueError as e:
+        if not coerce:
```

Comment:
Without the try/except this is the error. 
It seems like _convert_listlike() would have to accept coerce as an argument to fix it?
```
Traceback (most recent call last):
  File "/Users/vmd/Github/pandas_to_datetime_fix/pandas/tseries/tests/test_timeseries.py", line 4467, in test_day_not_in_month_coerce_true_NaT
    self.assertTrue(isnull(to_datetime('2015-02-29', format="%Y-%m-%d", coerce=True)))
  File "/Users/vmd/Github/pandas_to_datetime_fix/pandas/tseries/tools.py", line 354, in to_datetime
    return _convert_listlike(np.array([ arg ]), box, format)[0]
  File "/Users/vmd/Github/pandas_to_datetime_fix/pandas/tseries/tools.py", line 341, in _convert_listlike
    raise e
  File "/Users/vmd/Github/pandas_to_datetime_fix/pandas/tseries/tools.py", line 332, in _convert_listlike
    if com.is_datetime64_dtype(result) and box:
  File "/Users/vmd/Github/pandas_to_datetime_fix/pandas/core/common.py", line 2516, in is_datetime64_dtype
    tipo = _get_dtype_type(arr_or_dtype)
  File "/Users/vmd/Github/pandas_to_datetime_fix/pandas/core/common.py", line 2479, in _get_dtype_type
    raise ValueError('%r is not a dtype' % arr_or_dtype)
ValueError: None is not a dtype

----------------------------------------------------------------------
Ran 3 tests in 0.019s
```